[joy-ui][List] Add the `marker` prop (#39313)
mui/material-ui: docs/data/joy/components/list/list.md

If used, it will automatically change the semantic element from the default `<li>` to `<div>`.
:::

### Markers

> To add markers to each of the list items, change their CSS display to `display: "list-item"`.

Reviewer: ```suggestion
Use `display: list-item` on the list item to add markers to each one of them.
```

> It is recommended to set min-height of the list item to `auto` because the display `list-item` will no longer align the items vertically.

Reviewer: ```suggestion
Make sure to add the `min-height: auto` property to the list item.
That's because adding `display: list-item` means the items no longer align vertically.
```

{{"demo": "MarkerList.js"}}

> To see all the markers, check out the [MDN — list style type](https://developer.mozilla.org/en-US/docs/Web/CSS/list-style-type#values) page.

Reviewer: ```suggestion
Visit the [MDN list-style-type page](https://developer.mozilla.org/en-US/docs/Web/CSS/list-style-type#values) to learn about all the possible markers you can use.
```

If used, it will automatically change the semantic element from the default `<li>` to `<div>`.
:::

### Marker

> Use the `marker` prop with any valid [list-style-type](https://developer.mozilla.org/en-US/docs/Web/CSS/list-style-type) value, to add a marker to the list items.

Reviewer 2: ```suggestion
Use the `marker` prop with any valid [list-style-type](https://developer.mozilla.org/en-US/docs/Web/CSS/list-style-type) value to add a marker to the list items.
```

> Use `marker` prop to add a marker to the list items. The value can be any valid [list-style-type](https://developer.mozilla.org/en-US/docs/Web/CSS/list-style-type) CSS property.

Reviewer: ```suggestion
Use the `marker` prop with any valid [list-style-type](https://developer.mozilla.org/en-US/docs/Web/CSS/list-style-type) value, to add a marker to the list items.
```

Just a nudge here!

{{"demo": "MarkerList.js"}}

:::success

> For [numbered lists](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ol), change the List component to ` component="ol"`. 

Reviewer 2: ```suggestion
For [numbered lists](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ol), change the List component to `component="ol"`. 
```

> For [numbered list](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ol), you should render the ordered list of items by setting `component="ol"` prop on the List component.

Reviewer: ```suggestion
For [numbered lists](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ol), change the List component to ` component="ol"`. 
```

A bit shorter!
